packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -557,6 +558,35 @@ class DebugIosLLDBInit extends IosLLDBInit {
   BuildMode get buildMode => BuildMode.debug;
 }
 
+class CheckDevDependenciesIos extends CheckDevDependencies {
+  const CheckDevDependenciesIos();
+
+  @override
+  String get name => 'check_dev_dependencies_ios';
+
+  @override
+  List<Source> get inputs => <Source>[
+    ...super.inputs,
+    const Source.pattern(
+      '{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/ios.dart',
+    ),
+
+    // The generated Xcode properties file contains
+    // the FLUTTER_DEV_DEPENDENCIES_ENABLED configuration.
+    // This target should re-run whenever that value changes.
+    if (FlutterProject.current().isModule)
+      const Source.pattern('{PROJECT_DIR}/.ios/Flutter/Generated.xcconfig')
+    else
+      const Source.pattern('{PROJECT_DIR}/ios/Flutter/Generated.xcconfig'),
```

Comment:
Nit: I'd avoid hardcoded paths

```suggestion
    final FlutterProject flutterProject = FlutterProject.current();
    final String generatedConfigFilePath = flutterProject.ios.generatedXcodePropertiesFile.path.replaceFirst(
      flutterProject.directory.path,
      '{PROJECT_DIR}/',
    );
    return <Source>[Source.pattern(generatedConfigFilePath)];
```

Reply:
Updated!